RFC/src/RFC-0110_BaseNodes.md

```diff
@@ -1 +1,185 @@
+THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and
+"OPTIONAL" in this document are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
+
+## Disclaimer
+
+The purpose of this document and its content is for information purposes only and may be subject to change or update
+without notice.
+
+This document may include preliminary concepts that may or may not be in the process of being developed by the Tari
+community. The release of this document is intended solely for review and discussion by the community regarding the
+technological merits of the potential system outlined herein.
+
+## Goals
+
+This document describes the roles that [base node]s play in the Tari network and their general approach for doing so.
+
+## Related RFCs
+
+* [RFC-0100: Base layer](RFC-0100_BaseLayer.md)
+
+## Description
+
+### Broad requirements
+
+Tari Base Nodes form a peer-to-peer network for a proof-of-work based blockchain running the [Mimblewimble]
+protocol. The proof of work is performed via merge mining with Monero. Arguments for this design are
+presented [in the overview](RFC-0001_overview.md#proof-of-work).
+
+Tari Base Nodes MUST carry out the following tasks:
+
+* Validate all [Tari coin] [transaction]s.
+* Propagate valid transactions to peer nodes.
+* Validate all new [block]s received by checking that
+* Propagate validated new blocks to peer nodes.
+* Connect to peer nodes to catch up (sync) its blockchain state
+* Provide historical block information to peers that are syncing.
+
+Once the Digital Assets Network goes live, Base Nodes will also need to support the tasks described in
+[RFC-0300/Assets](RFC-0300_DAN.md). These requirements are omitted for the moment.
+
+To carry out these tasks effectively, Base Nodes SHOULD:
+
+* Save the [blockchain] into a indexed local database.
+* Maintain an index of all unspent outputs ([UTXO]s).
+* Maintain a list of all pending, valid transactions that have not yet been mined (the [mempool]).
+* Manage a list of Base Node peers present on the network.
+
+Tari Base Nodes MAY also implement the following services via an API to clients. Such clients may include "light"
+clients, block explorers, wallets, and Tari applications:
+
+  * Block queries
+  * Kernel data queries
+  * Transaction queries
+  * Submission of new transactions
+
+### Transaction validation and propagation
+
+Base nodes can notified of new transactions from
+* connected peers
+* clients via APIs
+
+When a new transaction has been received, it has the `unvalidated` [ValidationState]. The transaction is then passed
+to the transaction validation service, where its state will become one of `rejected` or `validated`.
+
+The transaction validation service checks that:
+
+* all inputs to the transaction are valid [UTXO]s
+* all inputs are not duplicated
```

Comment:
all inputs are not duplicated
vs.
no inputs are duplicated

Reply:
This statement "_all inputs are not duplicated_" use negative logic that can be easily misunderstood. Suggest changing it to "_no inputs are duplicated_" if that is the meaning, otherwise you have lost me too.